packages/js/components/src/calendar/date-picker.js

```diff
@@ -76,13 +76,21 @@ class DatePicker extends Component {
 	}
 
 	render() {
-		const { date, disabled, text, dateFormat, error, isInvalidDate } =
-			this.props;
+		const {
+			date,
+			disabled,
+			text,
+			dateFormat,
+			error,
+			isInvalidDate,
+			popoverProps = { inline: true },
```

Comment:
Let me share here a potential issue of moving the DOM element where the popover is rendered. I think the idea behind putting the elements completely out of anchor context is to make the component as much as possible isolated from the rest of the app.
It isn't a blocker from my PoV. Just a note.

Reply:
You're right, @retrofox. I agree that a long-term solution involves making the component look good and functional without inline rendering. This quick fix will enable users to use the component for now, but I'll create an issue to address that refactor during this cycle.

Reply:
I created [this issue](https://github.com/woocommerce/woocommerce/issues/41408) to tackle the components refactor.